eventmesh/ApiServerLockManager.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+const _ = require('lodash');
+const Promise = require('bluebird');
+const eventmesh = require('./');
+const errors = require('../common/errors');
+const EtcdLockError = errors.EtcdLockError;
+const logger = require('../common/logger');
+const CONST = require('../common/constants');
+const NotFound = errors.NotFound;
+const Conflict = errors.Conflict;
+
+class ApiServerLockManager {
+  /*
+  This method checks whether lock is of write type.
+  returns true if lock is present and not expired and it's of type WRITE
+  */
+  /**
+   * @param {string} resourceId - Id (name) of the resource that is being locked. In case of deployment lock, it is instance_id
+   */
+
+  isWriteLocked(resourceId) {
+    return eventmesh.server.getLockResourceOptions(CONST.APISERVER.RESOURCE_TYPES.LOCK, CONST.APISERVER.RESOURCE_NAMES.DEPLOYMENT_LOCKS, resourceId)
+      .then(options => {
+        const currentTime = new Date();
+        const lockDetails = JSON.parse(options);
+        const lockTime = new Date(lockDetails.lockTime);
+        const lockTTL = lockDetails.lockTTL ? lockDetails.lockTTL : Infinity;
+        if (lockDetails.lockType === CONST.ETCD.LOCK_TYPE.WRITE && ((currentTime - lockTime) < lockTTL)) {
+          logger.info(`Resource ${resourceId} was write locked for ${lockDetails.lockedResourceDetails.operation ?
+            lockDetails.lockedResourceDetails.operation : 'unknown'} ` +
+            `operation with id ${lockDetails.lockedResourceDetails.resourceId} at ${lockTime} `);
+          return true;
+        }
+        return false;
+      })
+      .catch(err => {
+        if (err instanceof NotFound) {
```

Comment:
Just handling `NotFound` in a filtered catch would be enough. No need to throw as well